source/game/field/RailInterpolator.hh

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <Common.hh>
+
+#include "game/field/Rail.hh"
+
+#include "game/system/map/MapdataPointInfo.hh"
+
+#include <egg/math/Vector.hh>
+
+namespace Field {
+
+class RailInterpolator {
+public:
+    RailInterpolator(f32 speed, u32 idx);
+    virtual ~RailInterpolator();
+
+    virtual void init(f32 t, u32 idx) = 0;
+    virtual u32 calc() = 0;
```

Comment:
```suggestion
    virtual Status calc() = 0;
```

This should return an enum class. These names aren't perfect, but just off of a cursory glance with what I've seen, some tentative suggestions:
```cpp
enum class Status {
    InProgress = 0,
    SegmentEnd = 1,
    ChangingDirection = 2,
};
```

Reply:
Honestly I think these names are fine for now at least.